### ISSUE TITLE
About detecting other browsers

### ISSUE THREAD

**Issue by SeYoungKim** [DamonOehlman/detect-browser#33]
If I use this library in other in-app browsers, I found that "browser" instance is "undefined". 
The browsers I tested on were "In-app browser in NaverApp mobile app" and "In-app browser in KakaoTalk app".
These are both mobile apps and very famous apps in Korea.
In NaverApp or kakaotalk(used in Korea), I got navigator.userAgent as follows.
(I used 

alert(navigator.appName + "\n" + navigator.appVersion + "\n" + navigator.userAgent + "\n" + navigator.appCodeName + "\n" + navigator.platform);

for testing. So please take a look at the underlined lines.)

![image](https://cloud.githubusercontent.com/assets/12039768/26528686/10884910-43eb-11e7-9282-d5c22cfbae64.png)

![image](https://cloud.githubusercontent.com/assets/12039768/26528693/31e2af92-43eb-11e7-96a8-a98fc23571df.png)

Because this module uses map and filter methods for the fixed browsers list, it cannot handle other cases.
  var browsers = [
    [ 'edge', /Edge\/([0-9\._]+)/ ],
    [ 'yandexbrowser', /YaBrowser\/([0-9\._]+)/ ],
    [ 'chrome', /(?!Chrom.*OPR)Chrom(?:e|ium)\/([0-9\.]+)(:?\s|$)/ ],
    [ 'crios', /CriOS\/([0-9\.]+)(:?\s|$)/ ],
    [ 'firefox', /Firefox\/([0-9\.]+)(?:\s|$)/ ],
    [ 'opera', /Opera\/([0-9\.]+)(?:\s|$)/ ],
    [ 'opera', /OPR\/([0-9\.]+)(:?\s|$)$/ ],
    [ 'ie', /Trident\/7\.0.*rv\:([0-9\.]+)\).*Gecko$/ ],
    [ 'ie', /MSIE\s([0-9\.]+);.*Trident\/[4-7].0/ ],
    [ 'ie', /MSIE\s(7\.0)/ ],
    [ 'bb10', /BB10;\sTouch.*Version\/([0-9\.]+)/ ],
    [ 'android', /Android\s([0-9\.]+)/ ],
    [ 'ios', /Version\/([0-9\._]+).*Mobile.*Safari.*/ ],
    [ 'safari', /Version\/([0-9\._]+).*Safari/ ],
  ];

What do you think about adding a logic for handling other user agent strings which are not caught by those predefined rules? like "default" in "switch".

**Comment by DamonOehlman** [comment]
@SeYoungKim Handling unknown browsers has come up before (see #30) and initially I thought that would be a good thing to do.  After actually writing a PR myself to add that functionality I considered that the current API is probably more correct (when you consider how things like flow / typescript might describe the API).

As such I did update the readme to show the following code example rather than the one that was there earlier (which I did think created some of this confusion):

```js
const browser = require('detect-browser');

// handle the case where we don't detect the browser
if (browser) {
  console.log(browser.name);
  console.log(browser.version);
}
```

As for adding the additional user agent strings to this is definitely something that can be done, and I'll add information to the repo (in the form of a contributing guide) about how to do this :)

**Comment by DamonOehlman** [comment]
@SeYoungKim I've added some notes on what a PR might contain to the README:

https://github.com/DamonOehlman/detect-browser#adding-additional-browser-support

I did go looking for a good example of a PR, however, couldn't find one.  I'm happy to use your case and make a reference PR myself if that would help you.  Is there some way that you can provide the text for those user agent strings rather than just the screenshots though?  Just a little worried I'll make a typo...

**Comment by DamonOehlman** [comment]
Fixed by #39 - big thanks to @5punk for that.
